packages/calcite-components/src/components/input/input.scss

```diff
@@ -612,6 +612,17 @@ input {
   }
 }
 
+:host([type="datetime-local"]) {
+  .element-wrapper {
+    inline-size: 100%;
+
+    input {
+      display: inline-block;
+      min-inline-size: 1px;
```

Comment:
@macandcheese, @driskull Should this be set to 0, for consistency? We could do this in a follow-up if there's agreement.